libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -581,17 +576,14 @@ internal object MapboxRouteLineUtils {
             route.directionsRoute.legs()?.forEachIndexed { legIndex, leg ->
                 val filteredIntersections = filterForRestrictedIntersections(leg)
                 ifNonNull(filteredIntersections) { stepIntersections ->
-                    val legDistancesArray =
-                        granularDistances?.nestedDistancesArray?.map { legDistances ->
-                            legDistances.flatten()
-                        }
+                    val legDistancesArray = granularDistances?.legsDistances
                     stepIntersections.forEach { stepIntersectionData ->
                         val geometryIndex = stepIntersectionData.first.geometryIndex()
                         if (geometryIndex != null && legDistancesArray?.isNotEmpty() == true) {
                             val distanceRemaining =
                                 legDistancesArray[legIndex][geometryIndex].distanceRemaining
-                            (1.0 - distanceRemaining / (granularDistances!!.distance)).apply {
-                                if (this < 1.0) {
+                            (1.0 - distanceRemaining / granularDistances!!.completeDistance).apply {
+                                if (this in 0.0..1.0) {
```

Comment:
There's no reason to add an expression stop at 1.0 since it's the end of the line. I'm concerned allowing 1.0 to go in could cause a problem with multi-leg routes.  I'm changing it to 0.9.

Reply:
1.0 would always be the end of the route, regardless of how many legs there are. I noticed that routes have often an injected intersection at the end of a route that never has a `restricted` class, so it made sense to me that this is correctly reflected in the expression. Otherwise, checks like these eventually become obscure.

What issue exactly are you concerned with? If you've seen a specific problem in the past, let's add a unit test for it.